src/commands/createFunction/durableSteps/sql/SqlServerCreateStep.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { Server, SqlManagementClient } from '@azure/arm-sql';
+import { AzExtLocation, LocationListStep } from '@microsoft/vscode-azext-azureutils';
+import { AzureWizardExecuteStep, ISubscriptionContext, nonNullProp, nonNullValue } from '@microsoft/vscode-azext-utils';
+import { Progress } from 'vscode';
+import { ext } from '../../../../extensionVariables';
+import { localize } from '../../../../localize';
+import { createSqlClient } from '../../../../utils/azureClients';
+import { ISqlDatabaseConnectionWizardContext } from '../../../appSettings/ISqlDatabaseConnectionWizardContext';
+
+export class SqlServerCreateStep<T extends ISqlDatabaseConnectionWizardContext> extends AzureWizardExecuteStep<T> {
+    public priority: number = 205;
+
+    public async execute(context: T & ISubscriptionContext, progress: Progress<{ message?: string; increment?: number }>): Promise<void> {
+        const client: SqlManagementClient = await createSqlClient(<T & ISubscriptionContext>context);
+        const rgName: string = nonNullValue(context.resourceGroup?.name);
+        const newServerName: string = nonNullProp(context, 'newSqlServerName');
+
+        const creating: string = localize('creatingSqlServer', 'Creating new SQL server "{0}"...', newServerName);
+        ext.outputChannel.appendLog(creating);
+        progress.report({ message: creating });
+
+        const location: AzExtLocation = await LocationListStep.getLocation(<T & ISubscriptionContext>context);
+        const serverOptions: Server = {
+            location: nonNullProp(location, 'name'),
+            administratorLogin: nonNullProp(context, 'newSqlAdminUsername'),
```

Comment:
Yup, it was the first method I tried to get working.  I've tried both methods.  They generate connection strings on the portal that you can use.  I originally tried using both of the generated strings, but only the SQL authentication method seemed to work.  I think I brought this up to you at the time when I first ran into this roadblock and you mentioned it was probably fine to just implement it this way for now.  I can look into implementing the Azure AD way again... but if I run into the same roadblock I might need to seek outside feedback on how to get that method running.

Reply:
Oh yeah, I vaguely remember that. Yeah, it's fine to have this way. I actually thought it would be _easier_ to not have to use SQL authentication, but if that's not the case, no worries.